components/admin/lessons/AdminLessonNav.tsx

```diff
@@ -0,0 +1,71 @@
+import { toUpper } from 'lodash'
+import React, { useState } from 'react'
+import { Tab, Col, Nav } from 'react-bootstrap'
+import styles from '../../../scss/adminLessonNav.module.scss'
+
+const AdminLessonNav = ({
+  navItems,
+  tabs
+}: {
+  navItems: { value: string; children?: string[] }[]
+  tabs: (() => JSX.Element)[]
+}) => {
+  const [key, setKey] = useState<string>('introduction')
+
+  const isActive = (k: string, data?: { active: string; inactive: string }) => {
+    const isActiveBool = k === key
+    if (!data?.active && !data?.inactive) return isActiveBool
+
+    return isActiveBool ? data.active : data.inactive
+  }
+
+  const renderItems = (items: { value: string; children?: string[] }[]) =>
+    items.map((item, index) => (
+      <Nav.Item key={index}>
+        <Nav.Link
+          active={isActive(item.value) as boolean}
+          className={
+            isActive(item.value, {
+              active: styles.lessons_tabsNav__nav__item,
+              inactive: styles['lessons_tabsNav__nav__item--inactive']
+            }) as string
+          }
+          eventKey={item.value}
+        >
+          {toUpper(item.value)}
+        </Nav.Link>
+      </Nav.Item>
+    ))
+
+  const renderTabPanes = (
```

Comment:
`renderItems` as well as this function shouldn't be functions. I don't see them being used more than once in this component, the function isn't passed down as a prop, and I notice these functions only return an array of jsx elements by using some higher order component.

Having them as functions complicates things -- in this context I think the code can be made more simple by just assigning the array of jsx elements to a variable directly, instead of having a function to do so.
```
navItems = items.map((item, index) => (
      <Nav.Item key={index}>
        <Nav.Link
          active={isActive(item.value) as boolean}
          className={
            isActive(item.value, {
              active: styles.lessons_tabsNav__nav__item,
              inactive: styles['lessons_tabsNav__nav__item--inactive']
            }) as string
          }
          eventKey={item.value}
        >
          {toUpper(item.value)}
        </Nav.Link>
      </Nav.Item>
))

tabPanes = items.map((item, index) => (
    <Tab.Pane key={index} eventKey={item.value}>
          {tabs[index]()}
    </Tab.Pane>
))
```

Reply:
I agree.